<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>mainWindow</class>
 <widget class="QMainWindow" name="mainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>750</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="mainLayout">
    <item>
     <widget class="QTabWidget" name="tabWidget">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="tabMonitor">
       <attribute name="title">
        <string>Monitor</string>
       </attribute>
       <layout class="QVBoxLayout" name="layoutMonitor">
        <item>
         <widget class="QGroupBox" name="groupBoxMonitor">
          <property name="title">
           <string>Monitor</string>
          </property>
          <layout class="QVBoxLayout" name="layoutMonitorGroupBox">
           <item>
            <widget class="QTableView" name="tableMonitor"/>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="groupBoxFilters">
          <property name="title">
           <string>Filters</string>
          </property>
          <layout class="QVBoxLayout" name="layoutFiltersGroupBox">
           <item>
            <widget class="QFrame" name="frameFilterInputs">
             <property name="frameShape">
              <enum>QFrame::NoFrame</enum>
             </property>
             <property name="frameShadow">
              <enum>QFrame::Raised</enum>
             </property>
             <layout class="QHBoxLayout" name="layoutFilterInputs">
              <property name="spacing">
               <number>6</number>
              </property>
              <property name="leftMargin">
               <number>0</number>
              </property>
              <property name="topMargin">
               <number>0</number>
              </property>
              <property name="rightMargin">
               <number>0</number>
              </property>
              <property name="bottomMargin">
               <number>0</number>
              </property>
              <item>
               <widget class="QPushButton" name="buttonAddFilter">
                <property name="text">
                 <string>Add</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLineEdit" name="inputFilterId">
                <property name="text">
                 <string/>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLineEdit" name="inputFilterByte0">
                <property name="text">
                 <string/>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLineEdit" name="inputFilterByte1"/>
              </item>
              <item>
               <widget class="QLineEdit" name="inputFilterByte2"/>
              </item>
              <item>
               <widget class="QLineEdit" name="inputFilterByte3"/>
              </item>
              <item>
               <widget class="QLineEdit" name="inputFilterByte4"/>
              </item>
              <item>
               <widget class="QLineEdit" name="inputFilterByte5"/>
              </item>
              <item>
               <widget class="QLineEdit" name="inputFilterByte6"/>
              </item>
              <item>
               <widget class="QLineEdit" name="inputFilterByte7"/>
              </item>
             </layout>
            </widget>
           </item>
           <item>
            <widget class="QFrame" name="frameFilterTable">
             <property name="frameShape">
              <enum>QFrame::NoFrame</enum>
             </property>
             <property name="frameShadow">
              <enum>QFrame::Raised</enum>
             </property>
             <layout class="QHBoxLayout" name="layoutFilterTable">
              <property name="leftMargin">
               <number>0</number>
              </property>
              <property name="topMargin">
               <number>0</number>
              </property>
              <property name="rightMargin">
               <number>0</number>
              </property>
              <property name="bottomMargin">
               <number>0</number>
              </property>
              <item>
               <widget class="QFrame" name="frameFilterButtons">
                <property name="frameShape">
                 <enum>QFrame::NoFrame</enum>
                </property>
                <property name="frameShadow">
                 <enum>QFrame::Raised</enum>
                </property>
                <property name="lineWidth">
                 <number>0</number>
                </property>
                <layout class="QVBoxLayout" name="layoutFilterButtons">
                 <property name="leftMargin">
                  <number>0</number>
                 </property>
                 <property name="topMargin">
                  <number>0</number>
                 </property>
                 <property name="rightMargin">
                  <number>0</number>
                 </property>
                 <property name="bottomMargin">
                  <number>0</number>
                 </property>
                 <item>
                  <widget class="QPushButton" name="buttonRemoveFilter">
                   <property name="text">
                    <string>Remove</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QPushButton" name="buttonClearFilters">
                   <property name="text">
                    <string>Clear</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <spacer name="filterButtonsSpacer">
                   <property name="orientation">
                    <enum>Qt::Vertical</enum>
                   </property>
                   <property name="sizeHint" stdset="0">
                    <size>
                     <width>20</width>
                     <height>28</height>
                    </size>
                   </property>
                  </spacer>
                 </item>
                </layout>
               </widget>
              </item>
              <item>
               <widget class="QTableWidget" name="tableFilters"/>
              </item>
             </layout>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="groupBoxMessages">
          <property name="title">
           <string>Messages</string>
          </property>
          <layout class="QVBoxLayout" name="layoutMessagesGroupBox">
           <item>
            <widget class="QFrame" name="frameMessageInputs">
             <property name="frameShape">
              <enum>QFrame::NoFrame</enum>
             </property>
             <property name="frameShadow">
              <enum>QFrame::Raised</enum>
             </property>
             <layout class="QHBoxLayout" name="layoutMessageInputs">
              <property name="spacing">
               <number>6</number>
              </property>
              <property name="leftMargin">
               <number>0</number>
              </property>
              <property name="topMargin">
               <number>0</number>
              </property>
              <property name="rightMargin">
               <number>0</number>
              </property>
              <property name="bottomMargin">
               <number>0</number>
              </property>
              <item>
               <widget class="QPushButton" name="buttonSendMessage">
                <property name="text">
                 <string>Send</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLineEdit" name="inputMessageId"/>
              </item>
              <item>
               <widget class="QLineEdit" name="inputMessageByte0">
                <property name="text">
                 <string/>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLineEdit" name="inputMessageByte1"/>
              </item>
              <item>
               <widget class="QLineEdit" name="inputMessageByte2"/>
              </item>
              <item>
               <widget class="QLineEdit" name="inputMessageByte3"/>
              </item>
              <item>
               <widget class="QLineEdit" name="inputMessageByte4"/>
              </item>
              <item>
               <widget class="QLineEdit" name="inputMessageByte5"/>
              </item>
              <item>
               <widget class="QLineEdit" name="inputMessageByte6"/>
              </item>
              <item>
               <widget class="QLineEdit" name="inputMessageByte7"/>
              </item>
             </layout>
            </widget>
           </item>
           <item>
            <widget class="QFrame" name="frameMessageName">
             <property name="frameShape">
              <enum>QFrame::NoFrame</enum>
             </property>
             <property name="frameShadow">
              <enum>QFrame::Raised</enum>
             </property>
             <layout class="QHBoxLayout" name="layoutMessageName">
              <property name="leftMargin">
               <number>0</number>
              </property>
              <property name="topMargin">
               <number>0</number>
              </property>
              <property name="rightMargin">
               <number>0</number>
              </property>
              <property name="bottomMargin">
               <number>0</number>
              </property>
              <item>
               <widget class="QPushButton" name="buttonAddMessage">
                <property name="text">
                 <string>Add</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLineEdit" name="inputMessageName"/>
              </item>
              <item>
               <spacer name="messageNameSpacer">
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>321</width>
                  <height>20</height>
                 </size>
                </property>
               </spacer>
              </item>
             </layout>
            </widget>
           </item>
           <item>
            <widget class="QFrame" name="frameMessageTable">
             <property name="frameShape">
              <enum>QFrame::NoFrame</enum>
             </property>
             <property name="frameShadow">
              <enum>QFrame::Raised</enum>
             </property>
             <layout class="QHBoxLayout" name="layoutMessageTable">
              <property name="leftMargin">
               <number>0</number>
              </property>
              <property name="topMargin">
               <number>0</number>
              </property>
              <property name="rightMargin">
               <number>0</number>
              </property>
              <property name="bottomMargin">
               <number>0</number>
              </property>
              <item>
               <widget class="QFrame" name="frameMessageButtons">
                <property name="frameShape">
                 <enum>QFrame::NoFrame</enum>
                </property>
                <property name="frameShadow">
                 <enum>QFrame::Plain</enum>
                </property>
                <property name="lineWidth">
                 <number>0</number>
                </property>
                <layout class="QVBoxLayout" name="layoutMessageButtons">
                 <property name="leftMargin">
                  <number>0</number>
                 </property>
                 <property name="topMargin">
                  <number>0</number>
                 </property>
                 <property name="rightMargin">
                  <number>0</number>
                 </property>
                 <property name="bottomMargin">
                  <number>0</number>
                 </property>
                 <item>
                  <widget class="QPushButton" name="buttonSendSelected">
                   <property name="text">
                    <string>Send</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QPushButton" name="buttonRemoveMessage">
                   <property name="text">
                    <string>Remove</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QPushButton" name="buttonClearMessages">
                   <property name="text">
                    <string>Clear</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <spacer name="messageButtonsSpacer">
                   <property name="orientation">
                    <enum>Qt::Vertical</enum>
                   </property>
                   <property name="sizeHint" stdset="0">
                    <size>
                     <width>20</width>
                     <height>10</height>
                    </size>
                   </property>
                  </spacer>
                 </item>
                </layout>
               </widget>
              </item>
              <item>
               <widget class="QTableWidget" name="tableMessages"/>
              </item>
             </layout>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tabRemote">
       <attribute name="title">
        <string>Remote</string>
       </attribute>
       <layout class="QGridLayout" name="layoutRemoteTab">
        <item row="0" column="0">
         <widget class="QGroupBox" name="groupBoxMode">
          <property name="title">
           <string>Mode</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout">
           <property name="sizeConstraint">
            <enum>QLayout::SetMinimumSize</enum>
           </property>
           <item>
            <widget class="QRadioButton" name="radioButtonServer">
             <property name="text">
              <string>Server</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QRadioButton" name="radioButtonClient">
             <property name="text">
              <string>Client</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QRadioButton" name="radioButtonLocal">
             <property name="text">
              <string>Local</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item row="0" column="2">
         <widget class="QGroupBox" name="groupBoxClient">
          <property name="title">
           <string>Client</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_2">
           <item row="0" column="0" alignment="Qt::AlignTop">
            <widget class="QLabel" name="labelClientIp">
             <property name="text">
              <string>IP</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1" alignment="Qt::AlignTop">
            <widget class="QLineEdit" name="inputClientIp">
             <property name="layoutDirection">
              <enum>Qt::LeftToRight</enum>
             </property>
             <property name="autoFillBackground">
              <bool>false</bool>
             </property>
            </widget>
           </item>
           <item row="3" column="0" alignment="Qt::AlignLeft|Qt::AlignBottom">
            <widget class="QPushButton" name="buttonClientConnect">
             <property name="text">
              <string>Connect</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QLineEdit" name="inputClientPort"/>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="labelClientPort">
             <property name="text">
              <string>Port</string>
             </property>
            </widget>
           </item>
           <item row="3" column="1" alignment="Qt::AlignRight|Qt::AlignBottom">
            <widget class="QFrame" name="frameClientStatus">
             <property name="frameShape">
              <enum>QFrame::NoFrame</enum>
             </property>
             <property name="frameShadow">
              <enum>QFrame::Raised</enum>
             </property>
             <layout class="QHBoxLayout" name="horizontalLayout_2">
              <property name="spacing">
               <number>0</number>
              </property>
              <property name="leftMargin">
               <number>0</number>
              </property>
              <property name="topMargin">
               <number>0</number>
              </property>
              <property name="rightMargin">
               <number>0</number>
              </property>
              <property name="bottomMargin">
               <number>0</number>
              </property>
              <item alignment="Qt::AlignRight|Qt::AlignBottom">
               <widget class="QLabel" name="labelClientStatus">
                <property name="text">
                 <string>Status:</string>
                </property>
               </widget>
              </item>
              <item alignment="Qt::AlignRight|Qt::AlignBottom">
               <widget class="QLabel" name="ledClientStatus">
                <property name="minimumSize">
                 <size>
                  <width>16</width>
                  <height>16</height>
                 </size>
                </property>
                <property name="maximumSize">
                 <size>
                  <width>16</width>
                  <height>16</height>
                 </size>
                </property>
                <property name="styleSheet">
                 <string notr="true">background-color: red; border-radius: 8px;</string>
                </property>
                <property name="text">
                 <string/>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item row="2" column="0">
         <spacer name="verticalSpacer_2">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="0" column="1">
         <widget class="QGroupBox" name="groupBoxServer">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="title">
           <string>Server</string>
          </property>
          <layout class="QGridLayout" name="gridLayout">
           <item row="0" column="8" alignment="Qt::AlignTop">
            <widget class="QLineEdit" name="inputServerPort"/>
           </item>
           <item row="0" column="1" alignment="Qt::AlignTop">
            <widget class="QLabel" name="labelServerPort">
             <property name="text">
              <string>Port</string>
             </property>
            </widget>
           </item>
           <item row="3" column="1" alignment="Qt::AlignLeft|Qt::AlignBottom">
            <widget class="QPushButton" name="buttonServerStart">
             <property name="enabled">
              <bool>true</bool>
             </property>
             <property name="sizePolicy">
              <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="text">
              <string>Start</string>
             </property>
            </widget>
           </item>
           <item row="3" column="8" alignment="Qt::AlignRight|Qt::AlignBottom">
            <widget class="QFrame" name="frameServerStatus">
             <property name="frameShape">
              <enum>QFrame::NoFrame</enum>
             </property>
             <property name="frameShadow">
              <enum>QFrame::Raised</enum>
             </property>
             <layout class="QHBoxLayout" name="horizontalLayout_4">
              <property name="spacing">
               <number>0</number>
              </property>
              <property name="leftMargin">
               <number>0</number>
              </property>
              <property name="topMargin">
               <number>0</number>
              </property>
              <property name="rightMargin">
               <number>0</number>
              </property>
              <property name="bottomMargin">
               <number>0</number>
              </property>
              <item alignment="Qt::AlignRight|Qt::AlignBottom">
               <widget class="QLabel" name="labelServerStatus">
                <property name="text">
                 <string>Status:</string>
                </property>
               </widget>
              </item>
              <item alignment="Qt::AlignRight|Qt::AlignBottom">
               <widget class="QLabel" name="ledServerStatus">
                <property name="minimumSize">
                 <size>
                  <width>16</width>
                  <height>16</height>
                 </size>
                </property>
                <property name="maximumSize">
                 <size>
                  <width>16</width>
                  <height>16</height>
                 </size>
                </property>
                <property name="styleSheet">
                 <string notr="true">background-color: red; border-radius: 8px;</string>
                </property>
                <property name="text">
                 <string/>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tabCAN">
       <attribute name="title">
        <string>CAN</string>
       </attribute>
       <layout class="QGridLayout" name="layoutCAN">
        <item row="0" column="0">
         <widget class="QGroupBox" name="groupBoxCANInterface">
          <property name="title">
           <string>CAN Interface</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_3">
           <item row="0" column="0">
            <widget class="QLabel" name="labelCanChannel">
             <property name="text">
              <string>Channel</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QLineEdit" name="inputCanChannel"/>
           </item>
           <item row="0" column="2">
            <widget class="QPushButton" name="buttonCanConnect">
             <property name="text">
              <string>Connect</string>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="labelCanBitrate">
             <property name="text">
              <string>Bitrate</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QLineEdit" name="inputCanBitrate"/>
           </item>
           <item row="1" column="2" alignment="Qt::AlignRight|Qt::AlignBottom">
            <widget class="QFrame" name="frameCanStatus">
             <property name="frameShape">
              <enum>QFrame::NoFrame</enum>
             </property>
             <property name="frameShadow">
              <enum>QFrame::Raised</enum>
             </property>
             <layout class="QHBoxLayout" name="horizontalLayout_3">
              <property name="spacing">
               <number>0</number>
              </property>
              <property name="leftMargin">
               <number>0</number>
              </property>
              <property name="topMargin">
               <number>0</number>
              </property>
              <property name="rightMargin">
               <number>0</number>
              </property>
              <property name="bottomMargin">
               <number>0</number>
              </property>
              <item alignment="Qt::AlignHCenter">
               <widget class="QLabel" name="labelCanStatus">
                <property name="text">
                 <string>Status:</string>
                </property>
               </widget>
              </item>
              <item alignment="Qt::AlignRight">
               <widget class="QLabel" name="ledCanStatus">
                <property name="minimumSize">
                 <size>
                  <width>16</width>
                  <height>16</height>
                 </size>
                </property>
                <property name="maximumSize">
                 <size>
                  <width>16</width>
                  <height>16</height>
                 </size>
                </property>
                <property name="styleSheet">
                 <string notr="true">background-color: red; border-radius: 8px;</string>
                </property>
                <property name="text">
                 <string/>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item row="0" column="1">
         <spacer name="horizontalSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>373</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="1" column="0">
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>547</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
